src/integration-tests/bash/cleanup.sh

```diff
@@ -523,32 +523,38 @@ if [ -x "$(command -v helm)" ]; then
   [[ $? == 1 ]] && HELM_VERSION=V3
   echo "Detected Helm Version [$(helm version --short --client)]"
   echo @@ `timestamp` Deleting installed helm charts
-  namespaces=`kubectl get ns | grep -v NAME | awk '{ print $1 }'`
-  for ns in $namespaces
-  do 
+  if [ "$HELM_VERSION" == "V2" ]; then
+   helm list --short | while read helm_name; do
    if [ ! "$DRY_RUN" = "true" ]; then
-     (
+   (
      set -x
-     helm list --short --namespace $ns | while read helm_name; do
-       if [ "$HELM_VERSION" == "V2" ]; then
-         helm delete --purge  $helm_name
-       else 
-         helm uninstall $helm_name -n $ns 
-       fi
-     done
-     )
+     helm delete --purge  $helm_name
+    )
    else
-     (
-     helm list --short --namespace $ns | while read helm_name; do
-       if [ "$HELM_VERSION" == "V2" ]; then
-         echo @@ `timestamp` Info: DRYRUN: helm delete --purge  $helm_name
+     echo @@ `timestamp` Info: DRYRUN: helm delete --purge  $helm_name
+   fi
+   done
+  fi
+
+  if [ "$HELM_VERSION" == "V3" ]; then
+   namespaces=`helm list --all-namespaces | grep -v NAME | awk '{print $2}'`
+   for ns in $namespaces
+   do 
+     echo "Looking for helm lists in namesapce $ns"
```

Comment:
The loop is still here

Reply:
namespaces variable will store only the namesapce(s) that has a helm deploymet

Reply:
The loop here is OK IMO, as long as we only iterate through each NS once per my previous comment.

Reply:
But if you want to further optimize, iterate through the output of the first all namespace command in a single "big loop" that parses its lines, which which parses out the ns from the line and parses out the helm name from the same line.